### PR TITLE
Add aws_lc_rs feature and set it as default

### PR DESCRIPTION
**Pull request by SajjadPourali**
This PR adds the aws_lc_rs feature to the instant-acme library, sets it as the default feature, and updates the dependencies.

### CODE REVIEW

**Comment by SajjadPourali** [comment]
To fix the test error, I disabled the test when both `ring` and `aws_lc_rs` features are enabled. The TLS client cannot choose a default provider when both are enabled.

See https://github.com/rustls/rustls/blob/main/rustls/src/crypto/mod.rs#L270-L283

**Comment by djc** [comment]
I'm happy to add a feature for using aws-lc-rs, but I want to stick with ring as the default for now and please avoid any changes unrelated to this (such as TOML reformatting and version bumps to other dependencies).

File: src/lib.rs [hunk 1]
```diff
@@ -13,14 +17,13 @@ use http_body_util::{BodyExt, Full};
 use hyper::body::{Bytes, Incoming};
 use hyper::header::{CONTENT_TYPE, LOCATION};
 use hyper::{Method, Request, Response, StatusCode};
+use hyper_util::client::legacy::{connect::Connect, Client as HyperClient};
```
Comment (djc) [inline src/lib.rs:20]: I prefer to stick to one line per module (so keeping the separate lines for `Connect` and `Client`, as before).

File: src/types.rs [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
-use ring::signature::{EcdsaKeyPair, KeyPair};
+#[cfg(all(feature = "ring", not(feature = "aws_lc_rs")))]
+pub(crate) use ring as ring_like;
+use ring_like::digest::{digest, Digest, SHA256};
+use ring_like::signature::{EcdsaKeyPair, KeyPair};
 use rustls_pki_types::CertificateDer;
 use serde::de::DeserializeOwned;
 use serde::ser::SerializeMap;
 use serde::{Deserialize, Serialize};
+use std::fmt;
```
Comment (djc) [inline src/types.rs:15]: This should remain in a separate block of imports by itself, as before.

File: Cargo.toml [hunk 3]
```diff
@@ -29,7 +33,7 @@ thiserror = "1.0.30"
 [dev-dependencies]
 anyhow = "1.0.66"
 clap = { version = "4.0.29", features = ["derive"] }
-rcgen = "0.12"
+rcgen = { version = "0.12", features = ["pem"], default-features = false }
```
Comment (djc) [inline Cargo.toml:36]: As elsewhere, please keep `default-features = false` before `features = []`.

**Review by djc — COMMENTED** [review]
That looks a lot better. There are still a few small unrelated changes that I'd prefer to avoid.

File: Cargo.toml [hunk 4]
```diff
@@ -12,15 +12,19 @@ keywords = ["letsencrypt", "acme"]
 categories = ["web-programming", "api-bindings"]
 
 [features]
-default = ["hyper-rustls"]
+default = ["hyper-rustls", "ring"]
+ring = ["dep:ring", "hyper-rustls?/ring", "rcgen/ring"]
+aws_lc_rs = ["dep:aws-lc-rs", "hyper-rustls?/aws-lc-rs", "rcgen/aws_lc_rs"]
```
Comment (djc) [inline Cargo.toml:17]: Would prefer to name the feature with hyphens instead of underscores.

**Comment by djc** [comment]
CI is failing because `--all-features` now enables `fips`, which only works on Linux. Probably makes sense to change CI to do the following:

- `check --no-default-features`
- `test --features hyper-rustls,aws-lc-rs`
- `test --features hyper-rustls,ring`

**Comment by SajjadPourali** [comment]
> CI is failing because `--all-features` now enables `fips`, which only works on Linux. Probably makes sense to change CI to do the following:
> 
>     * `check --no-default-features`
> 
>     * `test --features hyper-rustls,aws-lc-rs`
> 
>     * `test --features hyper-rustls,ring`

The suggested commands don’t work because ring is in the default feature.
What about the following?
* `cargo check`
* `cargo test`
* `cargo test --no-default-features --features hyper-rustls,aws_lc_rs`

**Comment by SajjadPourali** [comment]
> CI is failing because `--all-features` now enables `fips`, which only works on Linux. Probably makes sense to change CI to do the following:
> 
>     * `check --no-default-features`
> 
>     * `test --features hyper-rustls,aws-lc-rs`
> 
>     * `test --features hyper-rustls,ring`

The suggested commands don’t work because ring is in the default feature.
What about the following?
* `cargo check`
* `cargo test`
* `cargo test --no-default-features --features hyper-rustls,aws_lc_rs`

**Comment by djc** [comment]
Ah, sorry, that sounds good!

**Comment by SajjadPourali** [comment]
@djc Done. Apparently, on Windows, NASM is a dependency for aws_lc_rs. Any idea?

**Review by djc — APPROVED** [review]
This is a good start. Going to merge this and follow up in a separate PR.

**Pull request merged**